cmd/sovereignnode/incomingHeader/incomingHeaderHandlerFactory.go

```diff
@@ -0,0 +1,40 @@
+package incomingHeader
+
+import (
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	mainFactory "github.com/multiversx/mx-chain-go/factory"
+	"github.com/multiversx/mx-chain-go/process"
+
+	hasherFactory "github.com/multiversx/mx-chain-core-go/hashing/factory"
+	marshallerFactory "github.com/multiversx/mx-chain-core-go/marshal/factory"
+)
+
+func CreateIncomingHeaderProcessor(
```

Comment:
This should be part of either run type or core run type in the future, meaning it should be an interface probably.

Reply:
add comment.

Reply:
Do we have a task for this created?

Reply:
Created task `MX-15513`